rewriter/TEnum.cc

```diff
@@ -82,11 +82,11 @@ vector<unique_ptr<ast::Expression>> processStat(core::MutableContext ctx, ast::C
         return badConst(ctx, stat->loc, klass->loc);
     }
 
-    if (rhs->fun != core::Names::new_() && rhs->fun != core::Names::let()) {
+    if (rhs->fun != core::Names::selfNew() && rhs->fun != core::Names::let()) {
         return badConst(ctx, stat->loc, klass->loc);
     }
 
-    if (rhs->fun == core::Names::new_() && !rhs->recv->isSelfReference()) {
+    if (rhs->fun == core::Names::selfNew() && !ast::MK::isMagicClass(rhs->recv.get())) {
```

Comment:
One thing I had in mind when I suggested using `ast::MK` here was that we actually make a helper for like `ast::MK::isSelfNew` so that the Magic-ness of `self.new` is completely hidden. i.e., what if this entire `if` condition was a helper?

Reply:
I originally was using `ast::MK::isSelfNew()` here, but this condition is subtly different, as it's checking for the case where the method is `new`, and the receiver is not `Magic`.

Reply:
Ahhh I misread that, thanks!